File: src/lib-node/dir-load.js
function dir_load(x)
{
 check(is_str,x)
 
 const r=arr()
 
 for(const v of dir_read(x))
 {
  if(is_file(v))
   push(r,v)
  else if(is_dir(v))
  {
   const a=dir_load(v)
   
   append(r,a)
  }
  else
   stop()
 }
 
 return r
}

File: src/lib-node/dir-load.cs
fn dir_load x
 check is_str x
 
 let r arr
 
 forof dir_read x true
  if is_file v
   push r v
  elseif is_dir v
   let a dir_load v
   
   append r a
  else
   stop
 end
 
 ret r
end
